analysis/task_analysis_surface/first_level_analysis_ss.py

```diff
@@ -0,0 +1,365 @@
+    canonical_contrasts = dict([(column, contrast_matrix[i])
+                      for i, column in enumerate(design_matrix.columns)])
+    
+    
+    #initialize dictionary of contrasts desired for analysis of respective task
+    final_contrasts={}
+    
+    #initialize list of complex contrasts, which are combinations of the canonical contrasts
+    task_contrasts = []
+    
+    #complex contrasts for nback task (must include a '-' or '+' and be made up of canonical contrasts)
+    if task == 'nback':
+        task_contrasts = ['twoback-zeroback']
+    
+    #complex contrasts for mid task (must include a '-' or '+' and be made up of canonical contrasts)
+    if task == 'mid':
+        task_contrasts = ['HiRewCue-NeuCue', #high reward anticipation
+                          'LoRewCue-NeuCue', #low reward anticipation
+                          'HiLossCue-NeuCue', #high loss anticipation
+                          'LoLossCue-NeuCue', #low loss anticipation
+                          
+                          'HiWin-NeuHit', #high reward outcome cp. to neutral hit
+                          'LoWin-NeuHit', #low reward outcome cp. to neutral hit
+                          'HiWin-HiNoWin', #high reward outcome cp. to high reward miss
+                          'LoWin-LoNoWin', #low reward outcome cp. to low reward miss
+                          
+                          'HiLoss-NeuMiss', #high loss cp. to neutral miss
+                          'LoLoss-NeuMiss', #low loss cp. to neutral miss
+                          'HiLoss-AvoidHiLoss', #high loss cp. to avoid high loss
+                          'LoLoss-AvoidLoLoss', #low loss cp. to avoid low loss
+                          
+                          'HiRewCue+LoRewCue-NeuCue', #combined reward anticipation
+                          'HiLossCue+LoLossCue-NeuCue', #combined loss anticipation
+                          
+                          'HiWin+LoWin-NeuHit', #combined reward outcome cp. to neutral hit
+                          'HiWin+LoWin-HiNoWin-LoNoWin', #combined reward outcome cp. to combined reward miss
+                          
+                          'HiLoss+LoLoss-NeuMiss', #combined loss cp. to neutral miss
+                          'HiLoss+LoLoss-AvoidHiLoss-AvoidLoLoss' #combined loss cp. to combined avoid loss  
+                         ]
+    
+    if task == 'sst':
+        #image and arrow as combined regressors 
+        #MJ and N as combined regressors
+        task_contrasts = ['SuccStop-Go', #correct inhibition
+                          'UnsuccStop-Go', #incorrect inhibition
+                          'SuccStop-UnsuccStop' #successful inhibitory control  
+                         ]
+    
+    #for loop that creates complex contrasts based on the string names from the task_contrasts list created above
+    for task_contrast in task_contrasts:
+        #split complex contrast string into a list of canonical contrasts separated by '-' or '+' 
```

Comment:
this seems like a complex section of code. perhaps just spot check that the vectors are correct